mchf-eclipse/drivers/audio/cw/cw_gen.c

```diff
@@ -699,13 +815,42 @@ void CwGen_AddChar(ulong code)
 			{
 				DigiModes_TxBufferPutChar(result);
 			}
+			// HACK END
 
 			break;
 		}
 	}
 
-	// HACK END
-	UiDriver_TextMsgPutChar(result);
+	if (result != '*')
+	{
+		UiDriver_TextMsgPutChar(result);
+	}
+	else
+	{
+		for (int i = 0; i<CW_SIGN_CODES; i++)
+		{
+			if (cw_sign_codes[i] == code) {
+				result = '-';
+
+				UiDriver_TextMsgPutSign(cw_sign_chars[i]);
+
+				// FIXME: HACK HACK FOR RTTY TX TESTING
+				// We can use the same buffer for all digimodes
+				if ((ts.txrx_mode == TRX_MODE_TX && is_demod_rtty()) || ts.cw_text_entry) // Can rtty rely just on cw_text_entry here?
```

Comment:
The idea is, check if a mode is active which accepts CW based input during TX, is active. Right now this definition applies to RTTY only. To scale this, I would suggest to either use ts.cw_text_entry for that purpose, i.e. any mode which wants text entry has to set this when it is going to  tx. Or we write a function to tell us this dynamically. Or we use a single additional variable to distinguish between an text entry mode for things like recording buttons or menu items and the use during transmissions as direct source of input.

Reply:
I would see ts.cw_text_entry as a single variable to check in this if. All modes, which allow such input then set its value to true and are responsible for setting to false after exiting the mode. Makes interface cleaner.